src/common/verify/isValidAccessGrant.test.ts

```diff
@@ -112,6 +112,21 @@ describe("isValidAccessGrant", () => {
     expect(mockedFetch).toHaveBeenCalled();
   });
 
+  it("uses the provided fetch if any from plain JSON VC", async () => {
```

Comment:
Unfortunately, this test doesn't really assert anything on the value passed to the verify endpoint, so it's not a great candidate to test for the plain JSON VC. That's on me, there wasn't a good way for you to know that in advance.

I think you can remove this test altogether, as the other one in this file covers the same use-case, but in a more meaningful way.